pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -162,8 +162,8 @@ func (a Adapter) Push(parameters common.PushParameters) (err error) {
 }
 
 // DoesComponentExist returns true if a component with the specified name exists, false otherwise
-func (a Adapter) DoesComponentExist(cmpName string) bool {
-	return utils.ComponentExists(a.Client, cmpName)
+func (a Adapter) DoesComponentExist(cmpName string) (bool, error) {
+	return utils.ComponentExists(a.Client, cmpName), nil
```

Comment:
I think `utils.ComponentExists()` should throw the underlying error too.

Reply:
kube's `utils.ComponentExists()` client API is a special case. It throws an error if there is no deployment(component). So its going to be difficult to distinguish between an actual error and when there is no deployment. So thats the reason I haven't changed kube's `utils.ComponentExists()`

Reply:
But why return an error at all here for `func (a Adapter) DoesComponentExist` then if we're always returning nil?

Reply:
@johnmcollier  because its an interface and docker is returning special err codes since docker has multiple containers and `utils.ComponentExists()` is different for docker. So `odo delete` should be able to handle the hanging containers see explanation here https://github.com/openshift/odo/pull/3405#discussion_r446366050

Reply:
Maybe I can dig around more again for the kube client api and see if we can make it uniform. Open to ideas to handle consistency between docker client and kube client for the interface.

Reply:
Hmm, yeah. Good point

Reply:
I think I can make Kube and Docker `utils.ComponentExists()` consistent

replacing 

```
deployment, err := c.KubeClient.AppsV1().Deployments(c.Namespace).Get(name, metav1.GetOptions{})
```

with 

```
label := "component=" + name
deploymentList, err := c.KubeClient.AppsV1().Deployments(c.Namespace).List(metav1.ListOptions{
	LabelSelector: label,
})
```

Reply:
> kube's utils.ComponentExists() client API is a special case. It throws an error if there is no deployment(component). So its going to be difficult to distinguish between an actual error and when there is no deployment and thus it returns only a bool.

We can use the kerrors package to distinguish between the errors https://github.com/openshift/odo/blob/9a470629475bf6339dec637ee5ea8e326188a7fd/pkg/occlient/occlient.go#L3334

Reply:
Thanks, I will use that